includes/Classifai/Providers/HTTPClient.php

```diff
@@ -0,0 +1,375 @@
+	 *
+	 * @since x.x.x
+	 *
+	 * @param string $url     The API URL.
+	 * @param array  $options Additional query params.
+	 * @return array|WP_Error
+	 */
+	public function post( string $url = '', array $options = [] ) {
+		$options = wp_parse_args(
+			$options,
+			[
+				'timeout' => $this->get_default_timeout(),
+			]
+		);
+
+		/**
+		 * Filter the URL for the post request.
+		 *
+		 * @since x.x.x
+		 * @hook classifai_{provider}_api_request_post_url
+		 *
+		 * @param {string} $url The URL for the request.
+		 * @param {array} $options The options for the request.
+		 * @param {string} $this->feature The feature name.
+		 *
+		 * @return {string} The URL for the request.
+		 */
+		$url = apply_filters( $this->get_filter_prefix() . '_api_request_post_url', $url, $options, $this->feature );
+
+		/**
+		 * Filter the options for the post request.
+		 *
+		 * @since x.x.x
+		 * @hook classifai_{provider}_api_request_post_options
+		 *
+		 * @param {array} $options The options for the request.
+		 * @param {string} $url The URL for the request.
+		 * @param {string} $this->feature The feature name.
+		 *
+		 * @return {array} The options for the request.
+		 */
+		$options = apply_filters( $this->get_filter_prefix() . '_api_request_post_options', $options, $url, $this->feature );
+
+		$this->add_headers( $options );
+
+		/**
+		 * Filter the response from the provider for a post request.
+		 *
+		 * @since x.x.x
+		 * @hook classifai_{provider}_api_response_post
+		 *
+		 * @param {array|WP_Error} $response The API response.
+		 * @param {string} $url Request URL.
+		 * @param {array} $options Request body options.
+		 * @param {string} $this->feature Feature name.
+		 *
+		 * @return {array} API response.
+		 */
+		return apply_filters(
+			$this->get_filter_prefix() . '_api_response_post',
+			$this->get_result( wp_remote_post( $url, $options ) ), // phpcs:ignore WordPressVIPMinimum.Functions.RestrictedFunctions.wp_remote_get_wp_remote_get
+			$url,
+			$options,
+			$this->feature
+		);
+	}
+
+	/**
+	 * Get results from the response.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param array|WP_Error $response The API response.
+	 * @return array|WP_Error
+	 */
+	public function get_result( $response ) {
+		if ( is_wp_error( $response ) ) {
+			return $response;
+		}
+
+		return $this->parse_response( $response );
+	}
+
+	/**
+	 * Parse the response from the API.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param array $response The API response
+	 * @return array|WP_Error
+	 */
+	protected function parse_response( array $response ) {
+		$code = wp_remote_retrieve_response_code( $response );
+		$body = wp_remote_retrieve_body( $response );
+
+		// Error responses
+		if ( $code >= 400 ) {
+			$response_message = wp_remote_retrieve_response_message( $response );
+			$status_text      = $response_message ? $response_message : __( 'Unknown error', 'classifai' );
+
+			// Try to extract specific error message from the server response
```

Comment:
Former code had chance for improvement:
```php
$message = $json['error']['message'] ?? esc_html__( 'An error occurred', 'classifai' );
```

It should be checking on `$json['error']['message']` only if `$json['error']` isn't a string itself.

It's expected now for users to see much more details from providers on errors instead of former `An error ocurred` message only.